client/packages/programs/src/JsonForms/components/DateOfBirth.tsx

```diff
@@ -54,7 +54,7 @@ const UIComponent = (props: ControlProps) => {
     handleChange(estimatedPath, false);
   };
 
-  const onChangeAge = (newAge: number) => {
+  const onChangeAge = (newAge: number | string = '') => {
```

Comment:
this should be `const onChangeAge = (newAge: number | undefined)` to match the `NumericTextInput` which does not have string as a possible input for the `onChange` prop.

This method then has to handle the undefined possibility

Reply:
Yeah, the `NumericTextInput` now explicitly requires a `number | undefined` type, so all these cases where the component was having to handle empty strings as well as number have now been simplified :)